### PR TITLE
Fix example and documentation rendering

### PR DESCRIPTION
**Pull request by simon-perriard**
List of changes:

- fixed the example in the documentation of `starky/lib.rs` and make it runnable
- bumped `katex` version to the latest available (`v0.16.11`)
- added `config.toml` in the different `.cargo` to automate KaTeX rendering
- updated README 
- tentatively added `--no-deps` flag for documentation rendering on https://docs.rs/, honestly don't know if it will work. Hopefully will render with KaTeX and fix the math rendering of pages like https://docs.rs/plonky2/latest/plonky2/gates/coset_interpolation/struct.CosetInterpolationGate.html

### CODE REVIEW

File: .cargo/config.toml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+[build]
+rustdocflags = ["--html-in-header", ".cargo/katex-header.html"]
```
Comment (Nashtare) [inline .cargo/config.toml:2]: nit: could you add a whitespace here and in other similar files?

**Review by Nashtare — APPROVED** [review]
Thank you @simon-perriard for addressing this, and apologies for the delay. I've verified the display, and the rendering for the `CosetInterpolationGate` indeed looks much better, thanks!

**Review by muursh — APPROVED** [review]

File: plonky2/.cargo/config.toml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+[build]
+rustdocflags = ["--html-in-header", ".cargo/katex-header.html"]
```
Comment (muursh) [inline plonky2/.cargo/config.toml:2]:
```suggestion
rustdocflags = ["--html-in-header", ".cargo/katex-header.html"]

```

File: starky/.cargo/config.toml [hunk 3]
```diff
@@ -0,0 +1,2 @@
+[build]
+rustdocflags = ["--html-in-header", ".cargo/katex-header.html"]
```
Comment (muursh) [inline starky/.cargo/config.toml:2]:
```suggestion
rustdocflags = ["--html-in-header", ".cargo/katex-header.html"]

```

**Pull request merged**